Fix tags issue while alternating visual and text mode in the classic editor

renatho opened the pull request:
Fixes #128 and fixes #127

#### Changes proposed in this Pull Request:

* Fix some issues while alternating between visual and text mode in the classic editor. The main issue is that it was creating some new lines.
* The other issue mentioned is that the angle brackets were appearing as `&lt;`. I could not reproduce it, but I believe these changes will fix it as well.

The main idea of this solution was to restore the unfiltered codes after the `afterWpautop`.

* I also identified that `<p>` and `<br>` tags were being removed in the core. So I replaced it temporarily to `<wp-p>` and `<wp-br>` tags. It'll appear only in the visual mode if the user changes from the text mode.

#### Testing instructions:

* Install and active the classic editor.
* Create a new post with the SyntaxHighlighter shortcode `[code language="php"]` or `[html]`. Please, test mixing and not mixing other texts in the post.
* Also test using `<p>` and `<br />` tags inner your code.
* Make sure that the shortcodes are keeping the indentation and break lines correctly, as well the angle brackets. And the rest of the content continues being treated as before.
* If possible, also test using the WP 4.2.3 (where the errors happened too) to confirm everything is working well. I tested here and it seems to be working fine as well.

Commit:
Fix new lines appearing when alternating between visual and text mode

Commit:
Add replacer to preserve tags p and br

Commit:
Refactor to use only the code pieces unfiltered

Commit:
Refactor wpautop code

jom commented:
I did some testing on this today. When I switched back to `Visual` the tags were replaced, is that expected? It looks like it didn't save with them though.

<img width="1208" alt="Screen Shot 2020-05-04 at 2 27 17 PM" src="https://user-images.githubusercontent.com/68693/80970630-5df3c100-8e13-11ea-98b1-f08a09eb67a3.png">

renatho commented:
> I did some testing on this today. When I switched back to `Visual` the tags were replaced, is that expected? It looks like it didn't save with them though.

@jom, yes. It's expected because of that:

> I also identified that <p> and <br> tags were being removed in the core. So I replaced it temporarily to <wp-p> and <wp-br> tags. It'll appear only in the visual mode if the user changes from the text mode.

Unfortunately, I didn't find a way to prevent it from being removed unless to replace it with a temporary mark.

jom reviewed (approved):
Ahh, sorry I missed this. It works well! Nice reorganization, too.
